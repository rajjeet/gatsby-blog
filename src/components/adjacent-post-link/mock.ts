export const previous = {
  fields: { slug: '/previous/' },
  frontmatter: { title: 'Introducing the New React Concurrent Mode' },
};
export const next = {
  fields: { slug: '/next/' },
  frontmatter: { title: 'The Implications of React Concurrent Mode' },
};
