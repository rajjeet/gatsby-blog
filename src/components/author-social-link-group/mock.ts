import { TProps } from './types';

export const makeProps = (): TProps => ({
  className: '',
});

