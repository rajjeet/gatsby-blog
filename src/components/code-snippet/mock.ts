import { TProps } from './types';

export const makeProps = (): TProps => ({
  children: `
const mainReducer = (state, action) => {
  console.log('Before', state);
  console.log('Action', action);
  let result = {
    count: countReducer(state.count, action)      
  };
  console.log('After', result);
  return result
};      
`,
  fileName: 'index.js',
});
