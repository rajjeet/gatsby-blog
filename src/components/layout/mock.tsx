import React from 'react';
import { TProps } from './types';
import { createMockGatsbyImageSharpFluid } from '../../utils/testing';
import { siteMetadata } from '../../../gatsby-config';

export const makeProps = (): TProps => ({
  children: <div>Sample child</div>,
  data: {
    file: createMockGatsbyImageSharpFluid.file,
    site: siteMetadata,
  },
});

