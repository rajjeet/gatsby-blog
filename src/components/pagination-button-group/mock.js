export const makeProps = () => ({
  className: {}, currentPage: 1, numOfPages: 5, paginationSlug: '',
});
