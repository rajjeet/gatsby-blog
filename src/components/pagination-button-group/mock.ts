export const makeProps = ({ currentPage = 1, numOfPages = 5 } = {}): TProps => ({
  className: '', currentPage, numOfPages, paginationSlug: '',
});

export type TProps = {
  className: string;
  currentPage: number;
  numOfPages: number;
  paginationSlug: string;
};
