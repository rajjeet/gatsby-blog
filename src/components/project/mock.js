import { createMockGatsbyImageSharpFluid } from '../../utils/testing';

export const makeProps = () => ({
  heading: 'Sample Heading',
  link: '',
  description: 'This is a sample description',
  className: {},
  thumbnail: createMockGatsbyImageSharpFluid.file.childImageSharp.fluid,
});
