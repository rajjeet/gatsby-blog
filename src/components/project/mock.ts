import { createMockGatsbyImageSharpFluid } from '../../utils/testing';
import { TProps } from './types';

export const makeProps = (): TProps => ({
  heading: 'Sample Project',
  link: '',
  description: 'This is a sample project description',
  className: '',
  thumbnail: createMockGatsbyImageSharpFluid.file.childImageSharp.fluid,
});

