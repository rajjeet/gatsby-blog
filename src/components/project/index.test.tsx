import React from 'react';
import { render } from '@testing-library/react';
import { Project } from './index';
import { makeProps } from './mock';

describe('<Project />', () => {
  it('should render', () => {
    const { asFragment } = render(<Project {...makeProps()} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have a link', () => {
    const { getByLabelText, getByRole } = render(<Project {...makeProps()} />);
    expect(getByLabelText('See Sample Project')).toBeDefined();
    expect(getByRole('link')).toBeDefined();
  });

  it('should have a heading', () => {
    const { getByText } = render(<Project {...makeProps()} />);
    expect(getByText('Sample Project')).toBeDefined();
  });

  it('should have a thumbnail image', () => {
    const { getByAltText } = render(<Project {...makeProps()} />);
    expect(getByAltText(/Sample Project/i)).toBeDefined();
  });

  it('should have a description', () => {
    const { getByText } = render(<Project {...makeProps()} />);
    expect(getByText('This is a sample project description')).toBeDefined();
  });
});
