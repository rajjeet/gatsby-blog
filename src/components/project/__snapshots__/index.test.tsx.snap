// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Project /> should render 1`] = `
<DocumentFragment>
  <div
    class="sc-htpNat fjEtFy"
  >
    <a
      aria-label="See Sample Project"
      class="sc-bwzfXH foniet"
      href=""
    >
      <h3>
        Sample Project
      </h3>
      <div
        class="sc-bdVaJa cBDXGb gatsby-image-wrapper"
        style="position: relative; overflow: hidden;"
      >
        <div
          style="width: 100%; padding-bottom: 100%;"
        />
        <picture>
          <source
            sizes=""
            srcset=""
          />
          <img
            alt="Sample Project"
            loading="lazy"
            sizes=""
            src="https://via.placeholder.com/250x180"
            srcset=""
            style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
          />
        </picture>
        <noscript>
          &lt;picture&gt;&lt;source srcset="" /&gt;&lt;img loading="lazy" src="https://via.placeholder.com/250x180" alt="Sample Project" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
        </noscript>
      </div>
      <p>
        This is a sample project description
      </p>
    </a>
  </div>
</DocumentFragment>
`;
